### PR TITLE
ST-1901 add redis queue

### PR DESCRIPTION
**Pull request by kkdai**
Implement basic FIFO queue

### CODE REVIEW

File: queue.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	return redigo.Int(q.Do("RPUSH", key, value))
+}
+
+// Dequeue with key
+func (q *Queue) Dequeue(key string) (string, error) {
+	return redigo.String(q.Do("LPOP", key))
+}
+
+func (q *Queue) RemoveAll(key string) (string, error) {
+	return redigo.String(q.Do("LTRIM", key, 1, -99))
```
Comment (zyfdegh) [inline queue.go]: What about delete all elements by `DEL key` or `LTRIM key -1 0`?

File: queue_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	assert.NotZero(t, n2)
+
+	v1, err3 := q.Dequeue("test1")
+	assert.NoError(t, err3)
+	assert.Equal(t, "value1", v1)
+
+	v2, err4 := q.Dequeue("test1")
+	assert.NoError(t, err4)
+	assert.Equal(t, "value2", v2)
+}
```
Comment (zyfdegh) [inline queue_test.go]: What about separating the tests to different functions using testing suite?

**Review by zyfdegh — CHANGES_REQUESTED** [review]
LGTM, but some adjustments are needed

File: queue.go (hunk 1 above)
Comment (kkdai) [reply queue.go]: Updateded

File: queue_test.go (hunk 2 above)
Comment (kkdai) [reply queue_test.go]: updated

**Review by kkdai — COMMENTED** [review]
Updated, @zyfdegh @c9s  Please help to review

File: queue_test.go (hunk 2 above)
Comment (zyfdegh) [reply queue_test.go]: Thank you, LGTM

**Review by zyfdegh — APPROVED** [review]

**Pull request merged**